ecommerce/serializers.py

```diff
@@ -544,18 +544,27 @@ def create(self, validated_data):
                 payment_type=validated_data.get("payment_type"),
                 payment_transaction=validated_data.get("payment_transaction"),
             )
-            for coupon in range(validated_data.get("num_coupon_codes")):
-                coupon = models.Coupon.objects.create(
-                    coupon_code=validated_data.get("coupon_code", uuid4().hex),
-                    payment=payment,
+            coupons = []
+            versions = []
+            eligibilities = []
+            for _ in range(validated_data.get("num_coupon_codes")):
+                coupons.append(
+                    models.Coupon(
+                        coupon_code=validated_data.get("coupon_code", uuid4().hex),
+                        payment=payment,
+                    )
```

Comment:
Instead of the `for` loop, couldn't this just be written as:

```python
coupons = [
    models.Coupon(
        coupon_code=validated_data.get("coupon_code", uuid4().hex),
        payment=payment,
    ) for _ in range(validated_data.get("num_coupon_codes", 0))
]
```

If that works, feel free to apply the same pattern as you see fit to the bulk creation loops below